### PR TITLE
tests for oracles && some fixes for UniV3Oracle

### PR DESCRIPTION
**Pull request by armoking32**
Added tests for all Oracles.
Some fixes in UniV3Oracle.sol

### CODE REVIEW

File: test/oracles/MellowOracle.test.ts [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+            });
+
+            it("initializes MellowOracle version", async () => {
+                expect("1.0.0").to.be.eq(
+                    await this.mellowOracle.contractVersion()
+                );
+            });
+
+            // corresponds to https://github.com/mellow-finance/mellow-vaults/blob/main/deploy/0054_MellowOracle_univ3.ts#L14
+            it("does not initialize IUniV2Oracle", async () => {
```
Comment (AlexSaplin) [inline test/oracles/MellowOracle.test.ts:85]: edge cases

File: test/oracles/ChainlinkOracle.test.ts [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+                    this.weth.address,
+                    this.wbtc.address,
+                ].forEach(async (token) => {
+                    var isSupported = await this.chainlinkOracle.hasOracle(
+                        token
+                    );
+                    expect(isSupported).to.be.true;
+                });
+            });
+            it(`returns false if oracle is not supported`, async () => {
```
Comment (AlexSaplin) [inline test/oracles/ChainlinkOracle.test.ts:283]: edge cases

File: test/oracles/ChainlinkOracle.test.ts [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+                                this.weth.address,
+                                this.usdc.address,
+                                this.wbtc.address,
+                            ],
+                            [this.chainlinkEth, this.chainlinkUsdc]
+                        )
+                ).to.be.revertedWith(Exceptions.INVALID_VALUE);
+            });
+
+            it(`reverts with ${Exceptions.FORBIDDEN} b.o. sender is not with admin rights`, async () => {
```
Comment (AlexSaplin) [inline test/oracles/ChainlinkOracle.test.ts:252]: edge cases

File: test/oracles/ChainlinkOracle.test.ts [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+                    BigNumber.from(32)
+                );
+
+                const pricesX96 = pricesResult.pricesX96;
+                const safetyIndices = pricesResult.safetyIndices;
+                expect(pricesX96.length).to.be.eq(1);
+                expect(safetyIndices.length).to.be.eq(1);
+            });
+
+            it(`reverts with ${Exceptions.INVALID_VALUE} b.o. different lengths of arrays`, async () => {
```
Comment (AlexSaplin) [inline test/oracles/ChainlinkOracle.test.ts:237]: edge cases

File: test/oracles/ChainlinkOracle.test.ts [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+            });
+
+            it(`returns true for ChainlinkOracle interface (${CHAINLINK_ORACLE_INTERFACE_ID})`, async () => {
+                let isSupported = await this.chainlinkOracle.supportsInterface(
+                    CHAINLINK_ORACLE_INTERFACE_ID
+                );
+                expect(isSupported).to.be.true;
+            });
+
+            it("returns false when contract does not support the given interface", async () => {
```
Comment (AlexSaplin) [inline test/oracles/ChainlinkOracle.test.ts:209]: edge cases

**Review by AlexSaplin — CHANGES_REQUESTED** [review]
Remove all .only
Check spec format

File: test/oracles/ChainlinkOracle.test.ts [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+
+                    const pricesX96 = pricesResult.pricesX96;
+                    const safetyIndices = pricesResult.safetyIndices;
+                    expect(pricesX96.length).to.be.eq(0);
+                    expect(safetyIndices.length).to.be.eq(0);
+                });
+            });
+
+            describe("when one of tokens is zero", () => {
+                it("retruns empty response", async () => {
```
Comment (AlexSaplin) [inline test/oracles/ChainlinkOracle.test.ts:101]: returns

File: test/oracles/UniV2Oracle.test.ts [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+            it("initializes version", async () => {
+                expect("1.0.0").to.be.eq(
+                    await this.uniV2Oracle.contractVersion()
+                );
+            });
+        });
+
+        describe("#price", () => {
+            it("retruns correct response for [uscd, weth] pair", async () => {
```
Comment (AlexSaplin) [inline test/oracles/UniV2Oracle.test.ts:59]: correct response -> prices

File: test/oracles/UniV3Oracle.test.ts [hunk 8]
```diff
@@ -0,0 +1,283 @@
...
+            expect(safetyIndexes.length).to.be.eq(correctResultSize);
+            for (var i = 0; i < correctResultSize; i++) {
+                expect(correctPricesX96[i]).to.be.eq(pricesX96[i]);
+                expect(correctSafetyIndexes[i]).to.be.eq(safetyIndexes[i]);
+            }
+        };
+
+        describe("#addUniV3Pools", () => {
+            describe("when adding [weth, usdc] pools with fee = 500", () => {
+                it("returns correct response", async () => {
```
Comment (AlexSaplin) [inline test/oracles/UniV3Oracle.test.ts:247]: adds pools

**Review by AlexSaplin — CHANGES_REQUESTED** [review]

**Review by AlexSaplin — APPROVED** [review]
